rust/agama-lib/src/scripts/model.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) [2024] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use std::{
+    fs,
+    io::Write,
+    os::unix::fs::OpenOptionsExt,
+    path::{Path, PathBuf},
+    process,
+};
+
+use serde::{Deserialize, Serialize};
+
+use crate::transfer::Transfer;
+
+use super::ScriptError;
+
+#[derive(Debug, Clone, Copy, PartialEq, strum::Display, Serialize, Deserialize)]
+#[strum(serialize_all = "camelCase")]
+#[serde(rename_all = "camelCase")]
+pub enum ScriptsGroup {
+    Pre,
+    Post,
+}
+
+/// Represents a script to run as part of the installation process.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Script {
+    /// Script's name.
+    pub name: String,
+    /// Script's body. Either the body or the URL must be specified.
+    pub body: Option<String>,
+    /// URL to get the script from. Either the body or the URL must be specified.
+    pub url: Option<String>,
```

Comment:
this looks like bad representation in rust. I kind of expect here enum with either body or url content (which are just marked string). I am just not sure how hard it will be to express it in serde.

Reply:
Yes, good point. It should be doable by playing with enums and "flatten".